tests/internal/test_cache.py

```diff
@@ -0,0 +1,67 @@
+import pytest
+from meatie.internal.cache import Cache
+
+
+class TestCache:
+    @pytest.fixture
+    def cache(self) -> Cache:
+        return Cache(max_size=3)
+
+    def test_store_and_load(self, cache):
+        cache.store("key1", "value1", ttl=10)
+        assert cache.load("key1") == "value1"
+
+    def test_load_nonexistent(self, cache):
+        assert cache.load("nonexistent") is None
+
+    def test_expiration(self, cache):
+        # Mock time
+        current_time = 100
+        cache._time_provider = lambda: current_time
+
+        cache.store("key1", "value1", ttl=10)
+        assert cache.load("key1") == "value1"
+
+        # Advance time past expiration
+        current_time = 111
+        assert cache.load("key1") is None
+
+    def test_max_size(self):
+        cache = Cache(max_size=2)
+
+        # Store 3 items
+        cache.store("key1", "value1", ttl=10)
+        cache.store("key2", "value2", ttl=10)
+        cache.store("key3", "value3", ttl=10)
+
+        # Should only have 2 newest items
+        assert cache.load("key1") is None
+        assert cache.load("key2") == "value2"
+        assert cache.load("key3") == "value3"
```

Comment:
In the project I am using GIVEN/WHEN/THEN comments to structure the tests. Would you mind adding them?